checker/src/main/java/com/pushtorefresh/private_constructor_checker/PrivateConstructorChecker.java

```diff
@@ -138,16 +180,26 @@ private void check(Class clazz) {
 
         final Constructor<?> constructor = constructors[0];
 
-        if (constructor.getParameterTypes().length > 0) {
-            throw new AssertionError(clazz + " has non-default constructor with some parameters");
-        }
-
         constructor.setAccessible(true);
 
         if (!Modifier.isPrivate(constructor.getModifiers())) {
             throw new AssertionError("Constructor of " + clazz + " must be private");
         }
 
+        final Class<?>[] parameterTypes = constructor.getParameterTypes();
```

Comment:
Yup for now only with one private constructor. I want to guarantee that there's only one private constructor which takes a string as a parameter, which will be called from factory methods.